File: lightbite-api/app/Models/MenuItem.php
<?php

declare(strict_types=1);

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\SoftDeletes;

class MenuItem extends Model
{
    use SoftDeletes;

    protected $fillable = [
        'uuid', 'restaurant_id', 'category_id', 'name', 'description',
        'price_fils', 'image_path', 'is_available', 'prep_time_minutes', 'sort_order',
    ];

    protected function casts(): array
    {
        return ['is_available' => 'boolean'];
    }

    public function restaurant(): BelongsTo
    {
        return $this->belongsTo(Restaurant::class);
    }

    public function category(): BelongsTo
    {
        return $this->belongsTo(MenuCategory::class);
    }
}

File: lightbite-app/lib/features/customer/restaurant/domain/entities/menu_item.dart
/// A single menu item belonging to a restaurant category.
class MenuItem {
  const MenuItem({
    required this.uuid,
    required this.name,
    this.description,
    required this.price,
    this.imageUrl,
    required this.isAvailable,
  });

  final String uuid;
  final String name;
  final String? description;
  final double price;
  final String? imageUrl;
  final bool isAvailable;
}
